warehouse/db.py

```diff
@@ -187,3 +189,23 @@ def includeme(config):
 
     # Register our request.db property
     config.add_request_method(_create_session, name="db", reify=True)
+
+    # Set a custom JSON serializer for psycopg
+    def serialize_as_json(obj):
+        def serialize(obj):
+            if isinstance(obj, (list, tuple)):
+                # If it's a list or tuple, recursively serialize each element
+                return [serialize(item) for item in obj]
+            elif isinstance(obj, dict):
+                # If it's a dictionary, recursively serialize each value
+                return {key: serialize(value) for key, value in obj.items()}
+            elif hasattr(obj, "__dict__"):
+                # If the object has a __dict__ attribute, serialize its __dict__
+                return serialize(obj.__dict__)
+            else:
+                # For other types, use default serialization
+                return obj
+
+        return json.dumps(serialize(obj))
+
+    psycopg.types.json.set_json_dumps(serialize_as_json)
```

Comment:
Another thought came up!

Since we're using sqlalchemy, should we be overriding the psycopg library itself, or setting s custom serializer during engine instantiation instead?
Refs: 
https://docs.sqlalchemy.org/en/20/dialects/postgresql.html#sqlalchemy.dialects.postgresql.JSON
https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.json_serializer